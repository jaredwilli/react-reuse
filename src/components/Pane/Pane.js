import React from 'react';
import PropTypes from 'prop-types';

/** Pane component for the atom of a tabs component */
const Pane = props => (
    <div className="tab-pane">
        {props.children}
    </div>
);

Pane.propTypes = {
    /** The label of the pane */
    label: PropTypes.string.isRequired,

    /** The children or content of the pane */
    children: PropTypes.element.isRequired
};

export default Pane;
