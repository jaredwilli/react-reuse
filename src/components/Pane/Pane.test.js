import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Pane from './Pane';

describe('Pane', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<Pane label="Tab 1"><div>This is my tab 1 contents!</div></Pane>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
