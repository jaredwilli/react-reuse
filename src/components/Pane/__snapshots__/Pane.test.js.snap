// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pane renders the component correctly 1`] = `
<div
  className="tab-pane"
>
  <div>
    This is my tab 1 contents!
  </div>
</div>
`;
